### PR TITLE
feat: refactor some code and add logs for debug

### PR DESCRIPTION
**Pull request by pritamstyz4ever**
## Context

Adding some logs and using ES6 features
## Objective

This PR adds some log information, refactors some code and converts functions to async-await

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -308,41 +308,46 @@ class K8sExecutor extends Executor {
...
      * @method checkPodResponse
      * @param {Object}      resp    pod response object
-     * @returns {null}
+     * @param {String}      buidlId  buildId
+     * @returns {Promise}
      */
-    checkPodResponse(resp) {
-        logger.debug(`Build pod response: ${JSON.stringify(resp, null, 2)}`);
+    checkPodResponse(resp, buidlId) {
+        return new Promise((resolve, reject) => {
```
Comment (jithine) [inline index.js:315]: I don't see any async actions in this function, why make it a Promise ?

File: index.js [hunk 2]
```diff
@@ -308,10 +308,10 @@ class K8sExecutor extends Executor {
      * check build pod response
      * @method checkPodResponse
      * @param {Object}      resp    pod response object
-     * @returns {null}
+     * @param {String}      buidlId  buildId
      */
-    checkPodResponse(resp) {
-        logger.debug(`Build pod response: ${JSON.stringify(resp, null, 2)}`);
+    checkPodResponse(resp, buidlId) {
+        logger.debug(`Build ${buidlId} pod response: ${JSON.stringify(resp, null, 2)}`);
```
Comment (tkyi) [inline index.js:314]:
```suggestion
    checkPodResponse(resp, buildId) {
        logger.debug(`Build ${buildId} pod response: ${JSON.stringify(resp, null, 2)}`);
```

Comment (tkyi) [inline index.js:314]: buildId*

File: index.js [hunk 3]
```diff
@@ -320,27 +320,22 @@ class K8sExecutor extends Executor {
         const status = resp.body.status.phase.toLowerCase();
         const waitingReason = hoek.reach(resp.body, CONTAINER_WAITING_REASON_PATH);
 
-        logger.debug(`Build pod status: ${status}`);
-        logger.debug(`Build container waiting reason: ${waitingReason}`);
+        logger.info(`Build ${buidlId} pod status: ${status}`);
+        logger.info(`Build ${buidlId} container waiting reason: ${waitingReason}`);
```
Comment (tkyi) [inline index.js:324]:
```suggestion
        logger.info(`Build ${buildId} pod status: ${status}`);
        logger.info(`Build ${buildId} container waiting reason: ${waitingReason}`);
```

File: index.js (hunk 2 above)
Comment (pritamstyz4ever) [reply index.js:314]: oh thanks, :-)

**Review by jithine — APPROVED** [review]

**Pull request merged**